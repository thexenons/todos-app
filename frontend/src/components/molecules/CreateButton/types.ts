import { ButtonProps } from "../../atoms/Button/types";

export interface CreateButtonProps {
	disabled?: ButtonProps["disabled"];
}
