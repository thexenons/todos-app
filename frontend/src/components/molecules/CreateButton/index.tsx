import { FC } from "react";
import { MdAdd } from "react-icons/md";

import Button from "../../atoms/Button";
import { CreateButtonProps } from "./types";

const CreateButton: FC<CreateButtonProps> = ({ disabled = false, onClick }) => (
	<Button
		variant={Button.variants.contained}
		colorVariant={Button.colorVariants.primary}
		disabled={disabled}
		onClick={onClick}
	>
		<Button.Icon>
			<MdAdd />
		</Button.Icon>
		Create
	</Button>
);

export default CreateButton;
