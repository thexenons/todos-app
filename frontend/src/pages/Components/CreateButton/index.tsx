import { FC } from "react";

import CreateButton from "../../../components/molecules/CreateButton";
import * as SS from "../styled";

const CreateButtonComponenstPage: FC = () => (
	<SS.ComponentsListWrapper>
		<h1>CreateButton</h1>
		<SS.ComponentsRowWrapper>
			<CreateButton />
			<CreateButton disabled />
		</SS.ComponentsRowWrapper>
	</SS.ComponentsListWrapper>
);

export default CreateButtonComponenstPage;
