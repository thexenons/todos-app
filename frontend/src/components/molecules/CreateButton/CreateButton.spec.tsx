import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import CreateButton from ".";

describe("CreateButton", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<CreateButton />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
