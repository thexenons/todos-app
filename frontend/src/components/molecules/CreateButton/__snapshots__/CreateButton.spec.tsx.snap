// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateButton > Render 1`] = `
<div>
  <button
    class="styled__ButtonWrapper-sc-f8cd241f-0 bYSNlB"
  >
    <span
      class="styled__ButtonIconsWrapper-sc-f8cd241f-1 ctjBQt"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    Create
  </button>
</div>
`;
